### PR TITLE
[CLA Approved] feat: AMD -> ES6

### LINKED ISSUE #6975: Update remaining AMD files to ES6 classes.
#### Summary

Throughout the codebase, we have a number of files while are still using the AMD pattern. These should be updated to be ES6 classes instead.

### PR DESCRIPTION
**Pull request by LeoDog896**
Fixes #6975, finishing a step for switching to an alternative ES6-only bundler like Vite.

(This is currently drafted as not all tests pass - there was weird ES6/AMD conversion hacks going on in the codebase).

**This isn't a full refactor to ES6 classes and ESMified code**. I've converted a _few_ to classes when the files were small or when necessary (e.g. with the main MCT class), but I've tried to avoid many refactors and only focus on AMD -> ES6 conversion. If this PR is accepted, I'll probably lead up with a second PR to finish cleanup of the rest of the folder.

### CODE REVIEW

**Comment by LeoDog896** [comment]
Random tidbit but there seems to be some odd TS issue happening (there is a partial block by https://github.com/microsoft/TypeScript/issues/55172), but I'll try to find a workaround.

**Comment by LeoDog896** [comment]
Odd - that one failing playwright test seems to be very flaky. I ran it locally, and it failed the first time, and was successful the second time. Maybe it's a timing issue?

**Comment by ozyx** [comment]
Hi @LeoDog896!

Thanks for contributing to Open MCT! If you haven't already, please complete and submit a [Contributor License Agreement (CLA)](https://github.com/nasa/openmct/discussions/3821). This is required before we can approve/merge your code.

> Random tidbit but there seems to be some odd TS issue happening (there is a partial block by [microsoft/TypeScript#55172](https://github.com/microsoft/TypeScript/issues/55172)), so I'll need to switch some functions into anonymous lambdas in the meantime.

Yep, that test is a current known flake, so likely not due to these changes.

>  If this PR is accepted, I'll probably lead up with a second PR to finish cleanup of the rest of the folder.

I think that's a good idea. This will take a while to review, and it's probably best to take care of large changes like this in manageable chunks. While we're waiting for your CLA to go through, please let us know if you have any questions or if we can assist in any way. Thanks again for this work!

**Comment by ozyx** [comment]
Hi @LeoDog896 ,

Thanks for your patience. We're still waiting to hear back about your CLA. I'll follow up with the folks in charge of that to see if we can get a status.

In the meantime, there have been quite a few changes since this PR was opened. Could you give a shot at resolving the resulting conflicts and let us know if you run into any issues?

**Comment by LeoDog896** [comment]
Sure! There shouldn't bee too many issues with merging, I'll resolve it in around 2-3 business days.

**Comment by ozyx** [comment]
@LeoDog896 At long last, I've received a follow-up about your CLA being approved! I'll give a shot at bringing this PR up to speed as it's been sitting here for a while. Really appreciate your patience on this one.

**Comment by LeoDog896** [comment]
Alright, thanks! @ozyx, if you're not able to do it, then let me know - I completely forgot after saying my previous comment 😅

**Comment by LeoDog896** [comment]
@ozyx ready for review 👍

File: src/api/objects/object-utils.js [hunk 1]
```diff
@@ -20,163 +20,161 @@
...
-   * @param keyString
-   * @returns domainObject
-   */
-  function toNewFormat(model, keyString) {
-    model = JSON.parse(JSON.stringify(model));
-    model.identifier = parseKeyString(keyString);
-    if (model.composition) {
-      model.composition = model.composition.map(parseKeyString);
-    }
+  return [identifier.namespace.replace(/:/g, '\\:'), identifier.key].join(':');
```
[the remark this answers is not in the document]
Comment (LeoDog896) [reply src/api/objects/object-utils.js:102]: [Already in this file](https://github.com/nasa/openmct/blob/715a44864e92e97f982a21c80522dc2f07f3d687/src/api/objects/object-utils.js#L103), this PR isn't to blame.

File: src/plugins/autoflow/AutoflowTabularPlugin.js [hunk 2]
```diff
@@ -20,23 +20,23 @@
...
+    views.addProvider({
+      name: 'Autoflow Tabular',
+      key: 'autoflow',
+      cssClass: 'icon-packet',
+      description: 'A tabular view of packet contents.',
+      canView: function (d) {
+        return !options || options.type === d.type;
+      },
+      view: function (domainObject) {
+        return new AutoflowTabularView(domainObject, openmct, document);
```
[the remark this answers is not in the document]
Comment (LeoDog896) [reply src/plugins/autoflow/AutoflowTabularPlugin.js:38]: [Same as in master](https://github.com/nasa/openmct/blob/715a44864e92e97f982a21c80522dc2f07f3d687/src/plugins/autoflow/AutoflowTabularView.js#L38) - interesting catch, though - wonder if it was caused by the fact that this now uses ES6

File: src/plugins/autoflow/AutoflowTabularView.js [hunk 3]
```diff
@@ -20,96 +20,92 @@
...
+        const available = height - SLIDER_HEIGHT;
+        const rows = Math.max(1, Math.floor(available / ROW_HEIGHT));
+        data.rowCount = rows;
+      }.bind(this);
 
-  AutoflowTabularView.prototype = Object.create(VueView.default.prototype);
+      interval = setInterval(updateRowHeight, 50);
+      this.$nextTick(updateRowHeight);
+    }
+  });
```
[the remark this answers is not in the document]
Comment (LeoDog896) [reply src/plugins/autoflow/AutoflowTabularView.js:108]: Same as above.

**Comment by LeoDog896** [comment]
I'm not sure about the `object-utils` security issue mentioned above being a security issue, but 2 new (non-security) issues were caught because of the use of ES6.

**Comment by ozyx** [comment]
> I'm not sure about the `object-utils` security issue mentioned above being a security issue, but 2 new (non-security) issues were caught because of the use of ES6.

Yep, looks like the scanner is catching some more stuff now that we're using ES6. Awesome!

File: src/api/api.js [hunk 4]
```diff
@@ -20,57 +20,38 @@
...
+  MenuAPI: MenuAPI,
+  NotificationAPI: NotificationAPI,
+  ObjectAPI: ObjectAPI,
+  PriorityAPI: PriorityAPI,
+  StatusAPI: StatusAPI,
+  TelemetryAPI: TelemetryAPI,
+  TimeAPI: TimeAPI,
+  TypeRegistry: TypeRegistry,
+  UserAPI: UserAPI,
+  AnnotationAPI: AnnotationAPI
```
Comment (ozyx) [inline src/api/api.js:56]:
```suggestion
  ActionsAPI,
  CompositionAPI,
  EditorAPI,
  FaultManagementAPI,
  FormsAPI,
  IndicatorAPI,
  MenuAPI,
  NotificationAPI,
  ObjectAPI,
  PriorityAPI,
  StatusAPI,
  TelemetryAPI,
  TimeAPI,
  TypeRegistry,
  UserAPI,
  AnnotationAPI
```

File: src/plugins/summaryWidget/src/views/SummaryWidgetViewProvider.js [hunk 5]
```diff
@@ -1,38 +1,52 @@
...
-      priority: function (domainObject) {
-        if (domainObject.type === 'summary-widget') {
-          return Number.MAX_VALUE;
-        } else {
-          return DEFAULT_VIEW_PRIORITY;
-        }
-      }
-    };
-  }
+/*****************************************************************************
```
Comment (ozyx) [inline src/plugins/summaryWidget/src/views/SummaryWidgetViewProvider.js:1]: nice catch :)

File: src/plugins/plugins.js [hunk 6]
```diff
@@ -20,225 +20,155 @@
...
+import InspectorDataVisualization from './inspectorDataVisualization/plugin';
+import InspectorViews from './inspectorViews/plugin';
+import ObjectInterceptors from './interceptors/plugin';
+import ISOTimeFormat from './ISOTimeFormat/plugin';
+import LADTable from './LADTable/plugin';
+import LocalStorage from './localStorage/plugin';
+import LocalTimeSystem from './localTimeSystem/plugin';
+import MyItems from './myItems/plugin';
+import NewFolderAction from './newFolderAction/plugin';
+import * as Notebook from './notebook/plugin';
```
Comment (ozyx) [inline src/plugins/plugins.js:59]:
```suggestion
import { NotebookPlugin, RestrictedNotebookPlugin } from './notebook/plugin';
```

File: src/plugins/plugins.js [hunk 7]
```diff
@@ -20,225 +20,155 @@
...
-  plugins.OperatorStatus = OperatorStatus.default;
-  plugins.Gauge = GaugePlugin.default;
-  plugins.Timelist = TimeList.default;
-  plugins.InspectorViews = InspectorViews.default;
-  plugins.InspectorDataVisualization = InspectorDataVisualization.default;
+plugins.SummaryWidget = SummaryWidget;
+plugins.TelemetryMean = TelemetryMean;
+plugins.URLIndicator = URLIndicatorPlugin;
+plugins.Notebook = Notebook.NotebookPlugin;
+plugins.RestrictedNotebook = Notebook.RestrictedNotebookPlugin;
```
Comment (ozyx) [inline src/plugins/plugins.js:132]:
```suggestion
plugins.Notebook = NotebookPlugin;
plugins.RestrictedNotebook = RestrictedNotebookPlugin;
```

File: src/ui/registries/ViewRegistry.js [hunk 8]
```diff
@@ -20,255 +20,104 @@
...
-   * views.
-   * @private
-   */
-  ViewRegistry.prototype.getByVPID = function (vpid) {
-    return this.providers.filter(function (p) {
-      return p.vpid === vpid;
-    })[0];
-  };
-
-  /**
```
Comment (ozyx) [inline src/ui/registries/ViewRegistry.js:124]: Let's keep these comments

File: src/ui/registries/ToolbarRegistry.js [hunk 9]
```diff
@@ -20,102 +20,68 @@
...
+};
 
-    this.providers[key] = provider;
-  };
+/**
+ * @private
+ */
+ToolbarRegistry.prototype.getByProviderKey = function (key) {
+  return this.providers[key];
+};
```
Comment (ozyx) [inline src/ui/registries/ToolbarRegistry.js:89]: Let's keep these comments

File: src/MCT.js [hunk 10]
```diff
@@ -20,72 +20,194 @@
...
-  RemoveActionPlugin,
-  MoveActionPlugin,
-  LinkActionPlugin,
-  DuplicateActionPlugin,
-  ImportFromJSONAction,
-  ExportAsJSONAction,
-  Vue
-) {
+import EventEmitter from 'EventEmitter';
+import { createApp, markRaw } from 'vue/dist/vue.esm-bundler';
```
Comment (ozyx) [inline src/MCT.js:24]:
```suggestion
import { createApp, markRaw } from 'vue';
```

File: src/MCT.js [hunk 11]
```diff
@@ -376,20 +325,12 @@ define([
...
-        components: {
-          Layout: Layout.default
-        },
-        provide: {
-          openmct: Vue.markRaw(this)
-        },
-        template: '<Layout ref="layout"></Layout>'
-      });
+      const appLayout = createApp(Layout);
+      appLayout.provide('openmct', markRaw(this));
```
Comment (ozyx) [inline src/MCT.js:330]: nice

**Review by ozyx — CHANGES_REQUESTED** [review]
Really nice work! I've left a few suggestions, but this looks great overall. The tests are passing too! One thing I'd like to test is whether or not this changes how `openmct` is used as a dependency. I suspect it should remain very much the same, but I'd like to verify that before we go ahead with this.

**Comment by ozyx** [comment]
> Really nice work! I've left a few suggestions, but this looks great overall. The tests are passing too! One thing I'd like to test is whether or not this changes how `openmct` is used as a dependency. I suspect it should remain very much the same, but I'd like to verify that before we go ahead with this.

Testing against our testing environment with openmct as a dependency and have had no issues at all. Fantastic

File: src/MCT.js [hunk 12]
```diff
@@ -20,72 +20,194 @@
...
-  LinkActionPlugin,
-  DuplicateActionPlugin,
-  ImportFromJSONAction,
-  ExportAsJSONAction,
-  Vue
-) {
+import EventEmitter from 'EventEmitter';
+import { createApp, markRaw } from 'vue/dist/vue.esm-bundler';
+
+import api from './api/api';
```
Comment (ozyx) [inline src/MCT.js:26]: Let's import each api individually:

`import TimeAPI from ./api/time/TimeAPI;`

etc.

File: src/MCT.js [hunk 13]
```diff
@@ -20,72 +20,194 @@
...
+  selection = new Selection(this);
+
+  /**
+   * MCT's time conductor, which may be used to synchronize view contents
+   * for telemetry- or time-based views.
+   * @type {module:openmct.TimeConductor}
+   * @memberof module:openmct.MCT#
+   * @name conductor
+   */
+  time = new api.TimeAPI(this);
```
Comment (ozyx) [inline src/MCT.js:79]: Let's declare these in the `constructor()`:

`this.time = new TimeAPI(this);`

**Review by ozyx — CHANGES_REQUESTED** [review]
Couple additional suggestions

File: src/MCT.js [hunk 14]
```diff
@@ -20,72 +20,194 @@
...
+  branding = BrandingAPI;
+
+  /**
+   * MCT's annotation API that enables
+   * human-created comments and categorization linked to data products
+   * @type {module:openmct.AnnotationAPI}
+   * @memberof module:openmct.MCT#
+   * @name annotation
+   */
+  annotation = new api.AnnotationAPI(this);
```
Comment (ozyx) [inline src/MCT.js:205]: Looks like this is in here twice-- let's only keep the one in the constructor.

File: src/MCT.js (hunk 10 above)
Comment (LeoDog896) [reply src/MCT.js:24]: That was a hacky fix - Vue throws some sort of error when that isn't used (something about `createApp` being imported wrongly)

Comment (ozyx) [reply src/MCT.js:24]: @LeoDog896 I think we can fix this just by adding a webpack alias to `.webpack/webpack.common.js`:

`vue: 'vue/dist/vue.esm-bundler'`

**Review by ozyx — CHANGES_REQUESTED** [review]
This is looking great-- super excited to get this merged. Just [one minor configuration fix](https://github.com/nasa/openmct/pull/7029/files/8b84d41c64f75282b9dc6adbb411d71a12367ed1..3564a4730dbac2020250174dad8f41fcc66bef11#r1437197713)

File: src/MCT.js (hunk 10 above)
Comment (LeoDog896) [reply src/MCT.js:24]: That could work 👍

**Comment by LeoDog896** [comment]
> This is looking great-- super excited to get this merged. Just [one minor configuration fix](https://github.com/nasa/openmct/pull/7029/files/8b84d41c64f75282b9dc6adbb411d71a12367ed1..3564a4730dbac2020250174dad8f41fcc66bef11#r1437197713)

👍, anything else?

**Review by ozyx — APPROVED** [review]
LGTM

**Pull request merged**